cmd/manager/main.go

```diff
@@ -160,6 +149,29 @@ func main() {
 		os.Exit(1)
 	}
 
+	if tool.Options.EnableLease {
+		operatorNs, err := k8sutil.GetOperatorNamespace()
+		if err != nil {
+			if err == k8sutil.ErrNoNamespace || err == k8sutil.ErrRunLocal {
+				log.Info("Skipping lease; not running in a cluster.")
+			} else {
+				log.Error(err, "Failed to get operator namespace")
+				os.Exit(1)
+			}
+		} else {
+			log.Info("Starting lease controller to report status")
+			leaseUpdater := lease.NewLeaseUpdater(
+				generatedClient,
+				"policy-controller",
+				operatorNs,
+				lease.CheckAddonPodFunc(generatedClient.CoreV1(), operatorNs, "app=policy-framework"),
```

Comment:
Unlike iam and cert policy controller which only contains one container in the pod, policy-framework contains mulitple pods. So a health check function was used to check the status of entire pod instead.